.github/workflows/test.yml

```diff
@@ -82,3 +88,85 @@ jobs:
         if: ${{ !startsWith(matrix.go, '1.18.') && !startsWith(matrix.go, '1.19.') }}
         run: |
           go test -race -shuffle=on -v -count=10 ./...
+
+  freebsd:
+    strategy:
+      matrix:
+        os: [ macos-12 ]
+        go: ['1.18.10', '1.19.12', '1.20.7', '1.21.0']
+    name: Test with Go ${{ matrix.go }} on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run in freebsd
+        uses: vmactions/freebsd-vm@v0
+        with:
+          usesh: true
+          prepare: |
+            fetch https://go.dev/dl/go${{matrix.go}}.freebsd-amd64.tar.gz
+            rm -fr /usr/local/go && tar -C /usr/local -xf go${{matrix.go}}.freebsd-amd64.tar.gz
+            ln -s /usr/local/go/bin/go /usr/local/bin
+          run: |
+            # FreeBSD tests run within a virtualbox VM on macOS.
+            # vmactions/freebsd-vm only supports a single "step" where it
+            # brings down the VM at the end of the step, so all
+            # the commands to run need to be put into this single block.
+
+            echo "Running tests on $(uname -a) at $PWD"
+
+            # verify Go is available
+            go version
+
+            echo "=> go build"
```

Comment:
My understanding is that it is impossible to have separate names like the above regular tests do, as these steps are executed in a VM. Is this correct?

Reply:
As far as I can tell, it only supports one run unit so single named block: https://github.com/vmactions/freebsd-vm/blob/main/index.js#L163

Adding other `-name` blocks will run on the host OS (macOS).

Reply:
yes to your question

Reply:
Acknowledged. Thanks!